proposals/shared-everything-threads/Overview.md

```diff
@@ -526,38 +523,43 @@ constructor.
+a new `thread-local: bool` option.
+
+When `thread-local` is true, the resulting `WebAssembly.SharedFunction` is a "thread-local function
+wrapper." It behaves like a thread-local global initialized to `ref.null nofunc` on each thread, so
+if it is called immediately after being shared with another thread, it will behave as though calling
+`null`. `WebAssembly.SharedFunction` has a `set` method that sets the wrapped function on the
 current realm, and it throws an error if the current realm's wrapped function has already been set
-to a non-null value. The `WebAssembly.SharedFunction` constructor with the `realm-local` option
+to a non-null value. The `WebAssembly.SharedFunction` constructor with the `thread-local` option
 supplied implicitly calls `set`, so it cannot be called again on that original thread (unless the
 function was constructed wrapping `null` in the first place).
 
-The `realm-local` option composes with the `async` option from JSPI. A shared function wrapper that
+The `thread-local` option composes with the `async` option from JSPI. A shared function wrapper that
 is also async is async on every realm and thread.
 
-As an engine-internal optimization, setting a realm-local global to a realm-local function wrapper
+As an engine-internal optimization, setting a thread-local global to a thread-local function wrapper
 can skip a layer of indirection and set the global to refer directly to the bound function, since it
-will never change or be observed from another realm. This optimization only works if Wasm
-"thread-local" globals are actually realm-local in the Web embedding. True thread-local globals can
-be emulated on top of realm-local globals by ensuring that all realms on a thread trampoline through
-functions from a single realm to enter WebAssembly.
+will never change or be observed from another thread. This optimization only works because the
+dynamic contexts of thread-local globals and thread-local function wrappers are the same, i.e. an
+entire thread or agent.
+
+> Note: On the web, JavaScript always executes in a particular [realm]. An agent (roughly, a thread
+> of execution) may have multiple active realms within it, such as in the case of multiple iframes.
+> Each agent that executes JS has at least one realm, but may have more. Importantly, each realm has
+> its own copy of all JS built-ins and web APIs. For example, a main document's `Array` constructor
+> is a different function than an iframe's `Array` constructor.
+>
+> It would be possible for thread-local globals and function wrappers to actually be realm-local
+> instead, which would allow the functions to be re-bound on each realm. This would be consistent
+> with the way many things work on the Web, but would be less consistent with how shared memory
+> multithreading works in Wasm today, where functions are bound (i.e. imported) once per thread, not
+> once per realm.
```

Comment:
I wonder if it's worth calling out in the notes that the prototypes of the JS reflections of shared Wasm things remain realm-local, which is analogous to primitive object wrappers in JS (cf [string wrappers](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String#string_primitives_and_string_objects)).

Reply:
Sure, I'll note that here, too.